syntax = "proto3";

package auth;

// Google Authentication Service
service AuthService {
  rpc StartGoogleAuth(GoogleAuthRequest) returns (AuthResponse);
}

// Request containing the Google authorization code
message GoogleAuthRequest {
  string code = 1;
}

// Response containing JWT tokens + user info
message AuthResponse {
  string access_token = 1;
  string refresh_token = 2;
  string email = 3;
  string name = 4;
}
